### ISSUE TITLE
Python API

### ISSUE THREAD

**Issue by schmod** [fabfuel/ecs-deploy#45]
Have you given any consideration to creating a python package that (roughly) replicates the functionality of the CLI?

I've got a (stripped-down) fork that I've been using with AWS Lambda, and was wondering if there's any interest in incorporating those changes upstream.

**Comment by fabfuel** [comment]
Hi @schmod,

internally the CLI module already uses a Python API, which could be redesigned for simpler usage within a Python app. So yes, generally there is interest. Is your fork public?

Best
Fabian

**Comment by foragerr** [comment]
@fabfuel when you say _already uses a Python API_ do you mean boto3?
if so, then yes, providing a higher level api similar to ecs-deploy cli would be very useful.

its possible to change environment variables using boto3, but involves pulling taskdef, editing json locally and pushing it back up.

**Comment by fabfuel** [comment]
No, I don't mean boto3. There is a Python API in ecs deploy, which currently serves as the mapper between the CLI interface and boto3. Therefore it is currently designed to work with the CLI input e.g., but generally this works:

```
from ecs_deploy.ecs import DeployAction, EcsClient

client = EcsClient()
deployment = DeployAction(client, 'my-cluster', 'my-server')

task_definition = deployment.get_current_task_definition(deployment.service)
task_definition.set_environment([
    ['app', 'my-var', 'my-value']
])

new_task_definition = deployment.update_task_definition(task_definition)
deployment.deploy(new_task_definition)
```
